pinot-core/src/main/java/org/apache/pinot/core/transport/TableRouteInfo.java

```diff
@@ -0,0 +1,198 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.transport;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.apache.pinot.common.request.BrokerRequest;
+import org.apache.pinot.common.request.InstanceRequest;
+import org.apache.pinot.core.routing.ServerRouteInfo;
+import org.apache.pinot.core.routing.TimeBoundaryInfo;
+import org.apache.pinot.spi.config.table.TableConfig;
+
+
+/**
+ * Routing information for a table required to route plan fragments to servers.
+ * RequestMap is used to submit plan fragments to servers.
+ * Offline and realtime routing tables are used to keep track of which servers are executing the query.
+ * getUnavailableSegments() returns the segments that are not available when calculating the routing table.
+ * numPrunedSegmentsTotal() returns the number of segments that were pruned when calculating the routing table.
+ */
+public interface TableRouteInfo {
+
+  @Nullable
+  String getOfflineTableName();
+
+  @Nullable
+  String getRealtimeTableName();
```

Comment:
These 2 are not general abstraction, and doesn't apply to logical table

Reply:
Unfortunately the fact that these are null or not is a core requirement of the code in `BaseSingleStageBrokerRequestHandler`. So it does apply to logical tables

Reply:
For a logical table with `m` OFFLINE tables and `n` REALTIME tables, what should we return here?

Reply:
`LogicalTableRouteInfo` returns the logical table name with the right suffixes. These names are eventually used to create offline & realtime broker requests. 
Check this draft PR for the implementation of that file: https://github.com/apache/pinot/pull/15634/commits/b0859ebff7daeb07f4f4462d63c9899580565df0#diff-05f6cda0d6be5f155dfb376d0401fe69a2b01b1e1f45a891f65ded1201563ee6R192